expr/func_highestlowest.go

```diff
@@ -0,0 +1,69 @@
+package expr
+
+import (
+	"sort"
+
+	"github.com/grafana/metrictank/consolidation"
+
+	"github.com/grafana/metrictank/api/models"
+)
+
+type FuncHighestLowest struct {
+	in      GraphiteFunc
+	n       int64
+	fn      string
+	highest bool
+}
+
+func NewHighestLowestConstructor(fn string, highest bool) func() GraphiteFunc {
+	return func() GraphiteFunc {
+		return &FuncHighestLowest{fn: fn, highest: highest}
+	}
+}
+
+func (s *FuncHighestLowest) Signature() ([]Arg, []Arg) {
+	if s.fn != "" {
+		return []Arg{
+			ArgSeriesList{val: &s.in},
+			ArgInt{key: "n", val: &s.n},
+		}, []Arg{ArgSeriesList{}}
+	}
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+		ArgInt{key: "n", val: &s.n},
+		ArgString{key: "func", val: &s.fn, validator: []Validator{IsConsolFunc}},
+	}, []Arg{ArgSeriesList{}}
+}
+
+func (s *FuncHighestLowest) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncHighestLowest) Exec(cache map[Req][]models.Series) ([]models.Series, error) {
```

Comment:
I love how simple this implementation is :)

Reply:
easy PR!